File: services/halo4/models/request/get_recent_matches.go
package request

import (
	"github.com/branch-app/branch-mono-go/domain/xboxlive"
	"github.com/branch-app/branch-mono-go/services/halo4/models/response"
)

type RecentMatches struct {
	Identity xboxlive.IdentityLookup `json:"identity" bson:"identity"`

	GameModeID response.GameMode `json:"gameModeId" bson:"gameModeId"`
	StartAt    uint              `json:"startAt" bson:"startAt"`
	Count      uint              `json:"count" bson:"count"`
}

File: dotnet/Apps/ServiceHalo4/App/GetRecentMatches.cs
using System.Threading.Tasks;
using Branch.Packages.Contracts.ServiceHalo4;
using Branch.Packages.Contracts.Common.Branch;
using Branch.Packages.Contracts.ServiceIdentity;
using AutoMapper;
using Branch.Packages.Enums.Halo4;

namespace Branch.Apps.ServiceHalo4.App
{
	public partial class Application
	{
		public async Task<ResGetRecentMatches> GetRecentMatches(IdentityRequest identReq, GameMode gameMode, uint startAt, uint count)
		{
			var identity = await _identityClient.GetXboxLiveIdentity(new ReqGetXboxLiveIdentity
			{
				Type = identReq.Type,
				Value = identReq.Value,
			});

			var response = await _waypointClient.GetRecentMatches(identity.ToIdentity(), gameMode, startAt, count);
			var result = Mapper.Map<ResGetRecentMatches>(response.recentMatches);
			result.CacheInfo = response.cacheInfo;

			return result;
		}
	}
}
